src/eval.rs

```diff
@@ -638,7 +638,7 @@ fn reduce_with_witness<F: LurkField>(
                 } else if head == store.sym("cdr") {
                     let (arg1, end) = match store.car_cdr_mut(&rest) {
                         Ok((car, cdr)) => (car, cdr),
-                        Err(_) => unreachable!("Invalid tag"), //TODO: double check
+                        Err(_) => panic!("Invalid tag"),
```

Comment:
If we're panicking because of an error, we might as well display the error itself rather than discard and make assumptions about its content.